client/my-sites/patterns/components/pattern-preview/index.tsx

```diff
@@ -89,6 +91,21 @@ function PatternPreviewFragment( {
 		copyButtonText = isPreviewLarge ? 'Pattern copied!' : 'Copied';
 	}
 
+	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
+	const patternType =
+		typeof patternTypeFilter !== 'undefined'
+			? getTracksPatternType( patternTypeFilter )
+			: patternTypeFilter;
+	const recordCopyEvent = ( tracksEventName: string ) => {
+		recordTracksEvent( tracksEventName, {
+			name: pattern?.name,
+			category,
+			type: patternType,
+			user_is_dev_account: isDevAccount ? '1' : '0',
+			view: isGridView ? 'grid' : 'list',
+		} );
+	};
```

Comment:
```suggestion

		const onCopy = ( pattern: Pattern | null ) => {
		recordTracksEvent( 'calypso_pattern_library_copy', {
			name: pattern?.name,
			category,
			type: patternType,
			user_is_dev_account: isDevAccount ? '1' : '0',
			view: isGridView ? 'grid' : 'list',
		} );
	};
```

To address Fredrik's feedback here https://github.com/Automattic/wp-calypso/pull/88875/files#r1539042934, We'll also need to update this function.

Reply:
Nevermind, looking at the various PR comments, looks like we'll eventually be introducing a new helper function for tracking library events so we don't have to sort out these properties over and over again. Ignore my feedback :p